src/samples/coap/lwm2m-client.c

```diff
@@ -1058,20 +1217,55 @@ main(int argc, char *argv[])
     struct security_obj_instance_ctx *security_data;
     struct server_obj_instance_ctx *server_data;
     int r;
+    enum sol_lwm2m_security_mode sec_mode = SOL_LWM2M_SECURITY_MODE_NO_SEC;
+    char *cli_name = NULL;
+    char usage[256];
 
     srand(time(NULL));
-    if (argc < 2) {
-        fprintf(stderr, "Usage: ./lwm2m-sample-client <client-name> [bootstrap]\n");
+
+    snprintf(usage, sizeof(usage), "Usage: ./lwm2m-sample-client <client-name> [-b] [-s SEC_MODE]\n"
+        "Where Factory Bootstrap is default and SEC_MODE is an integer as per:\n"
+        "\tPRE_SHARED_KEY=%d\n"
+        "\tRAW_PUBLIC_KEY=%d\n"
+        "\tCERTIFICATE=%d\n"
+        "\tNO_SEC=%d (default)\n",
+        SOL_LWM2M_SECURITY_MODE_PRE_SHARED_KEY,
+        SOL_LWM2M_SECURITY_MODE_RAW_PUBLIC_KEY,
+        SOL_LWM2M_SECURITY_MODE_CERTIFICATE,
+        SOL_LWM2M_SECURITY_MODE_NO_SEC);
+
+    while ((r = getopt(argc, argv, "bs:")) != -1) {
+        switch (r) {
+        case 'b':
+            data_ctx.is_bootstrap = true;
+            break;
+        case 's':
+            sec_mode = atoi(optarg);
+            if (sec_mode < 0 || sec_mode > 3) {
+                fprintf(stderr, "%s", usage);
```

Comment:
It's up there on the `usage` string itself: https://github.com/solettaproject/soletta/pull/2272/files#diff-31f229684ffba7b5bad2c6f184b0afd2R1231.